lib/src/iterable_extensions.dart

```diff
@@ -914,6 +914,18 @@ extension IterableIterableExtension<T> on Iterable<Iterable<T>> {
       };
 }
 
+/// Extensions that apply to iterables of tuples/pairs (records with two unnamed elements).
+extension IterableTupleExtension<T1, T2> on Iterable<(T1, T2)> {
+  /// Returns a map from the first element of each tuple to the second element.
+  Map<T1, T2> get toMap => {for (final element in this) element.$1: element.$2};
```

Comment:
(Just as style comments, not endorsing the functionality: 
`toX` functionality like `toMap()` is usually a method, not a getter.
The `for` loop can destructure: `{for (var (key, value) in this) key: value}`.
This package doesn't use `final` for local variables.)